thrust/thrust/system/cuda/detail/async/copy.h

```diff
@@ -132,7 +132,7 @@ auto async_copy_n(
       thrust::raw_pointer_cast(&*output)
     , thrust::raw_pointer_cast(&*first)
     , sizeof(T) * n
-    , direction_of_copy(from_exec, to_exec)
+    , direction_of_copy(from_exec, to_exec).value
```

Comment:
That is actually one of the few use cases where it is ok to use the call operator:
1. It is a runtime value and not a compile time one
2. It is only in C++14 upwards, so it will continue to compile So we can keep that as is

Reply:
In that case I will revert this, and then `integral_constant` should be good to go I believe.